### PR TITLE
Add feature flags and merge Additional Expenses WIP PR

### PR DESCRIPTION
**Pull request by compwron**
### What github issue is this PR for, if any?
Related to #2760 but does not finish it

### What changed, and why?
Add the bones of additional expense. Add feature flagging (no library, just the db table and service- to be used from rails console for now, eventually from All Casa Admin UI)

What are feature flags? https://martinfowler.com/articles/feature-toggles.html https://en.wikipedia.org/wiki/Feature_toggle

### How will this affect user permissions?
no

### How is this tested? (please write tests!) 💖💪
feature flags are tested with rspec. Additional Expenses are not thouroughly tested yet. 

### Screenshots please :)
Screen Shot 2022-01-04 at 7 45 30 PM
Screen Shot 2022-01-04 at 7 45 27 PM

### CODE REVIEW

File: app/controllers/case_contacts_controller.rb [hunk 1]
```diff
@@ -88,12 +88,16 @@ def edit
...
   def update
+    # binding.pry
     authorize @case_contact
     @casa_cases = [@case_contact.casa_case]
     @selected_cases = @casa_cases
     @current_organization_groups = current_organization.contact_type_groups
 
     if @case_contact.update_cleaning_contact_types(update_case_contact_params)
+      additional_expense_params.each { |single_additional_expense_params|
+        @case_contact.additional_expenses.create(single_additional_expense_params)
```
Comment (compwron) [inline app/controllers/case_contacts_controller.rb:99]: this is the cool new bug we added where it creates a new additional expense every time you try to EDIT it. Let's fix this 😂

Comment (compwron) [reply app/controllers/case_contacts_controller.rb:99]: we can make `additional_expense_params` allow `id` and then use `update_or_create_by`

Comment (DrewAPeterson7671) [reply app/controllers/case_contacts_controller.rb:101]: Ok, I see the issue and I remember we talked about it last week.  This isn't working so far, but my initial approach is something like:
```
  def create_case_contact_for_every_selected_casa_case(selected_cases)
    selected_cases.map { |casa_case|
      ActiveRecord::Base.transaction do
        case_contact = casa_case.case_contacts.create(create_case_contact_params)
        if case_contact.persisted? && additional_expense_params&.any?
          additional_expense_params.each { |single_additional_expense_params|
            additional_expense_params = AdditionalExpense.exists?(id: additional_expense_params.dig("id")) ? AdditionalExpense.where(id: additional_expense_params.dig("id")).update(single_additional_expense_params) : AdditionalExpense.create(single_additional_expense_params)
            # case_contact.additional_expenses.create(single_additional_expense_params)
          }
        end
        case_contact
      end
    }
  end
```
Please let me know if you see something that makes it a bad idea (and, of course, it isn't currently working :-) )

**Comment by compwron** [comment]
```
rspec ./spec/controllers/case_contacts_controller_spec.rb:156 # CaseContactsController POST #create with valid params with additional expense creates additional expense
rspec ./spec/requests/case_contacts_spec.rb:110 # /case_contacts logged in as volunteer PATCH /update with valid parameters updates the requested case_contact and redirects to the root path
rspec ./spec/system/case_contacts/edit_spec.rb:11 # case_contacts/edit when admin admin successfully edits case contact
rspec ./spec/system/case_contacts/edit_spec.rb:45 # case_contacts/edit volunteer user is successful
```

File: app/controllers/case_contacts_controller.rb (hunk 1 above)
Comment (compwron) [reply app/controllers/case_contacts_controller.rb:101]: Deeply nested but functional, let's try it
And let's write tests :)

Comment (DrewAPeterson7671) [reply app/controllers/case_contacts_controller.rb:101]: I'm not finding an update_or_create method.  I did find a first_or_create that can be chained to an update.  However (just FYI) I put a binding.pry in that each block and it never triggers.  It seems its not going through that block at all.  So I'm scratching my head on that and poking around with it.

Comment (compwron) [reply app/controllers/case_contacts_controller.rb:101]: @DrewAPeterson7671 the method is https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by

**Review by DrewAPeterson7671 — APPROVED** [review]

**Pull request merged**